steengaardPAVisitor.cpp

```diff
@@ -4,503 +4,510 @@
+    } else {
+      recordVarAccess(lhs->getExprLoc(),
+                      std::pair<unsigned long, AccessType>(lhs_pair.first, WR));
+      traverse_subExpr(rhs);
+    }
+  }
+}
+
+bool SteengaardPAVisitor::VisitStmt(Stmt *st) {
+  LineCount++;
+  if (clang::BinaryOperator *bop = dyn_cast<clang::BinaryOperator>(st))
+    updatePAonBinaryExpr(bop);
+  else if (clang::UnaryOperator *uop = dyn_cast<clang::UnaryOperator>(st))
+    updatePAonUnaryExpr(uop);
+
+  // Update PA for function calls
+  // if x=f(Args), this st is processed by both updatePAOnCallExpr and
+  // updatePAonBinaryExpr
   else if (CallExpr *call = dyn_cast<CallExpr>(st)) {
-    updatePAOnCallExpr(call,-1);
+    updatePAOnCallExpr(call, -1);
   }
 
   else if (ReturnStmt *rtst = dyn_cast<ReturnStmt>(st)) {
-    if(rtst->getRetValue() && isVisitingFunc){
-      Expr *retexpr=rtst->getRetValue()->IgnoreImplicit();
-      unsigned long id=current_fs->rets[0];
-      updatePABasedOnExpr(id,retexpr);
+    if (rtst->getRetValue() && isVisitingFunc) {
+      Expr *retexpr = rtst->getRetValue()->IgnoreImplicit();
+      unsigned long id = current_fs->rets[0];
+      updatePABasedOnExpr(id, retexpr);
     }
-    if(isVisitingFunc) isVisitingFunc=false;
-  }  
-  else if (IfStmt *ifstmt = dyn_cast<IfStmt>(st)) {
-    Expr *ifexp=ifstmt->getCond();
-    if(ifexp) traverse_subExpr(ifexp->IgnoreImplicit());
-  }  
-  else if (ForStmt *forstmt = dyn_cast<ForStmt>(st)) 
-  {
-    Expr *forexp=forstmt->getCond();
-    if(forexp) traverse_subExpr(forexp->IgnoreImplicit());
+    if (isVisitingFunc)
+      isVisitingFunc = false;
+  } else if (IfStmt *ifstmt = dyn_cast<IfStmt>(st)) {
+    Expr *ifexp = ifstmt->getCond();
+    if (ifexp)
+      traverse_subExpr(ifexp->IgnoreImplicit());
+  } else if (ForStmt *forstmt = dyn_cast<ForStmt>(st)) {
+    Expr *forexp = forstmt->getCond();
+    if (forexp)
+      traverse_subExpr(forexp->IgnoreImplicit());
+  } else if (WhileStmt *whilestmt = dyn_cast<WhileStmt>(st)) {
+    Expr *whileexp = whilestmt->getCond();
+    if (whileexp)
+      traverse_subExpr(whileexp->IgnoreImplicit());
   }
-  else if(WhileStmt *whilestmt = dyn_cast<WhileStmt>(st)) 
+  return true;
+}
+
+std::set<clang::FunctionDecl *>
+SteengaardPAVisitor::getCallsFromFuncPtr(CallExpr *call) {
+  auto children = call->getRawSubExprs();
+  llvm::ArrayRef<Stmt *>::iterator stmt_it = children.begin();
+  std::set<FunctionDecl *> funcCalls;
+  Stmt *S = *stmt_it;
+  if (auto *S_as_expr = dyn_cast<Expr>(S))
   {
-    Expr *whileexp=whilestmt->getCond();
-    if(whileexp) traverse_subExpr(whileexp->IgnoreImplicit());
+    S = S_as_expr->IgnoreImplicit();
   }
-   return true;
- }
-
-
- std::set<FunctionDecl *> SteengaardPAVisitor::getCallsFromFuncPtr(CallExpr *call)
- {
-   ArrayRef<Stmt*> children=call->getRawSubExprs();
-   ArrayRef<Stmt*>::iterator I=children.begin();
-   std::set<FunctionDecl *> funcCalls;
-   Stmt *S=(*I)->IgnoreImplicit();
```

Comment:
Stmt::IgnoreImplicit was removed in llvm commit e64aee87a0cd5e7146e4dbba68072deecb8a4739. Suggestion is to use it through clang::Expr pointer, so I added a dyn_cast